src/cpp/lpnamer/main/ProblemGenerationExeOptions.cpp

```diff
@@ -21,19 +23,33 @@ ProblemGenerationExeOptions::ProblemGenerationExeOptions()
                            "use this option if unnamed problems are provided");
 }
 void ProblemGenerationExeOptions::Parse(unsigned int argc,
-                                        const char *const *argv) {
+                                        const char* const* argv) {
   OptionsParser::Parse(argc, argv);
   auto log_location = LOGLOCATION;
-  using namespace std::string_literals;
-  if (!XpansionOutputDir().empty() && !ArchivePath().empty()) {
-    auto msg = "Giving both archive and output options is not permitted"s;
-    throw ProblemGenerationOptions::ConflictingParameters(msg, log_location);
-  }
-  if (XpansionOutputDir().empty() && ArchivePath().empty()) {
-    auto msg = "Both output directory and archive path are empty"s;
+  checkAtLeastOneMandatoryOption(log_location);
+  checkOnlyOneMandatoryOption(log_location);
+}
+void ProblemGenerationExeOptions::checkAtLeastOneMandatoryOption(
+    const std::string& log_location) const {
+  if (std::vector<std::string> mandatory = {this->XpansionOutputDir().string(),
+                                            this->ArchivePath().string(),
+                                            this->StudyPath().string()};
+      std::ranges::all_of(
+          mandatory, [](std::string_view string) { return string.empty(); })) {
+    auto msg =
+        "Need to give at least on of [OutputDir, Archive, Study] options"s;
     throw ProblemGenerationOptions::MissingParameters(msg, log_location);
   }
 }
+void ProblemGenerationExeOptions::checkOnlyOneMandatoryOption(
+    const std::string& log_location) const {
+  if (std::vector<std::string> args = {this->XpansionOutputDir().string(),
+                                       this->ArchivePath().string(), this->StudyPath().string()};
+      std::ranges::count_if(args, std::ranges::empty) < (args.size() - 1)) {
```

Comment:
Why `args.size() - 1` and not `2` as it seems that the condition does not really depend on the number of elements in `args` ?

Reply:
Indeed, looking at the code, `args.size()` is always 3 regardless of it's content. I agree that writing 2 instead of `args.size() - 1` would make the code more clear.

Reply:
The rational was that if a new parameter is introduced, ards.size()-1 becomes 3 so instead of making 2 changes (list of args and size of args) using args.size()-1 made it that only changing the list was necessary.
But it's now reworled